src/Microsoft.Data.SqlClient/tests/ManualTests/ProviderAgnostic/ReaderTest/ReaderTest.cs

```diff
@@ -24,8 +24,13 @@ public static void TestMain()
             DbProviderFactory provider = SqlClientFactory.Instance;
             try
             {
-                using (DbConnection con = provider.CreateConnection())
+                using (SqlConnection con = (SqlConnection) provider.CreateConnection())
                 {
+                    // Testing in macOS Azure SQL container requires access token authentication
+                    if (!DataTestUtility.AuthenticatingWithoutAccessToken)
+                    {
+                        con.AccessToken = DataTestUtility.AADAccessToken;
+                    }
```

Comment:
If you introduce an extension method your change will be simpler without repeating yourself.

```suggestion
                using (DbConnection con = provider.CreateConnectionEx())
```

```c#
static DbConnection CreateConnectionEx(this DbProviderFactory provider)
{
    DbConnection conn = provider.CreateConnection();

    // Testing in macOS Azure SQL container requires access token authentication
    if (!DataTestUtility.AuthenticatingWithoutAccessToken && conn is SqlConnection sqlConn)
    {
        sqlConn.AccessToken = DataTestUtility.AADAccessToken;
        return sqlConn;
    }
    return conn;
}
```